### PR TITLE
Refactor conditions to properly support +/- infinity

### LINKED ISSUE #429: Poněkud šílené logování intervalů
Třeba tady:

    MarketplaceFilter (#5: [LoanInterestRateCondition (Range: <0.09490000; 179769313486231570000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.00000000>.)

Sice to není problém, je to jen log. A není to špatně, je to jen hnusné. Musí to zmizet.

### PR DESCRIPTION
**Pull request by triceo**
This closes #429 and also brings a better version of #431 to master.

@jhrcek Would you mind running your fancy strategy test suite against this code? You should experience exactly 0 changes - from an outside point of view, the code should behave exactly the same as before.

### CODE REVIEW

**Comment by jhrcek** [comment]
@triceo ok, I executed the integration tests and got some failures.

```
Caused by: java.lang.IllegalArgumentException: Threshold 1.01000000 does not fit Domain{clz=class com.github.robozonky.api.Ratio, minimum=0.00000000, maximum=1.00000000}
	at com.github.robozonky.strategy.natural.conditions.RangeCondition.relativeLessThan(RangeCondition.java:91)
	at com.github.robozonky.strategy.natural.conditions.RelativeLoanTermCondition.lessThan(RelativeLoanTermCondition.java:29)
	at com.github.robozonky.strategy.natural.NaturalLanguageStrategyParser.relativeTermConditionRangeClosedRight(NaturalLanguageStrategyParser.java:4701)
```

```
Caused by: java.lang.IllegalArgumentException: Threshold 85 does not fit Domain{clz=class java.lang.Integer, minimum=0, maximum=84}
	at com.github.robozonky.strategy.natural.conditions.RangeCondition.lessThan(RangeCondition.java:45)
	at com.github.robozonky.strategy.natural.conditions.LoanTermCondition.lessThan(LoanTermCondition.java:28)
	at com.github.robozonky.strategy.natural.NaturalLanguageStrategyParser.termConditionRangeClosedRight(NaturalLanguageStrategyParser.java:4505)
```

Most likely because there's now a bit stricter validation? I'll look into why this is failing today or tomorrow at the latest. I suspect this will be something like off-by-one errors in the testing strategy generator..

**Comment by triceo** [comment]
True, the domain is now validated - so, if we expect at most 84 months, the value of 85 will fail.
Likewise in the first stack trace - we expect values of 100 % at most.

File: robozonky-strategy-natural/src/main/antlr4/imports/CommonFilters.g4 [hunk 1]
```diff
@@ -82,17 +81,17 @@ termCondition returns [MarketplaceFilterCondition result]:
...
 
 termConditionRangeClosedLeft returns [MarketplaceFilterCondition result]:
     MORE_THAN min=intExpr
-    { $result = new LoanTermCondition($min.result + 1); }
+    { $result = LoanTermCondition.moreThan($min.result); }
 ;
 
 termConditionRangeClosedRight returns [MarketplaceFilterCondition result]:
     LESS_THAN max=intExpr
-    { $result = new LoanTermCondition(0, $max.result - 1); }
```
Comment (jhrcek) [inline robozonky-strategy-natural/src/main/antlr4/imports/CommonFilters.g4]: I think it's these `+ - 1`s that make some of the existing integration tests for the web tool fail. For example previous parser accepted condition like this:
`Ignorovat vše, kde: délka nedosahuje 85 měsíců.`

Anyway after tweaking those +1s in the generator, the tests are passing fine. If you're ok with that I'll push the changes in my integration tests.

Comment (triceo) [reply robozonky-strategy-natural/src/main/antlr4/imports/CommonFilters.g4]: This change is intentional - before this change, the conditions represented ranges closed on both sides. (`<A, B>`) To use an example, a way to make sure a range doesn't accept a number `B` was to specify it as `<A, B - 1>`.

With the new code, ranges are now either "less than A" `(-inf; A)`, "more than B" `(B; +inf)` or "anything between C and D" `<C; D>`. So, the behavior you are seeing is the expected behavior.

Feel free to merge your changes, I'll merge this PR. Thanks for checking, @jhrcek !

**Pull request merged**